dsa/Cargo.toml

```
@@ -22,12 +22,12 @@ opaque-debug = "0.3"
 pkcs8 = { version = "0.9", default-features = false, features = ["alloc"] }
 rand = { version = "0.8", default-features = false }
 rfc6979 = { version = "0.3", path = "../rfc6979" }
-signature = { version = ">= 1.5, < 1.7", default-features = false, features = ["digest-preview", "rand-preview"] }
+sha2 = "0.10"
```

This is the cause of the build failure. It enables `std` by default: https://github.com/RustCrypto/hashes/blob/1731ced/sha2/Cargo.toml#L30

It succeeded before because the check is `cargo build` which ignores `dev-dependencies`.

You need to do:

```suggestion
sha2 = { version = "0.10", default-features = false }
```

---

I applied this suggestion, though first locally.